plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.page.ts

```diff
@@ -200,10 +201,28 @@ class ProductCollectionPage {
 		}
 	}
 
+	async chooseProductInEditorProductPicker(
+		pageReference: Page | FrameLocator
+	) {
+		const editorProductPicker = pageReference.locator(
+			SELECTORS.productPicker
+		);
+		if ( await editorProductPicker.count() ) {
+			await editorProductPicker
+				.locator( 'label' )
+				.filter( {
+					hasText: 'Album',
+				} )
+				.click();
+		}
```

Comment:
I'm afraid this might become unstable as the `locator.count()` returns the currently available count, regardless of whether the nodes are still loading (no auto-waiting). In other words, if called too early in the rendering process, it will return `0`. It's important to remember that we must be as deterministic as possible. If we can avoid conditional actions like this, we should. Is there any other (UI) way we can determine whether the picker is available?

In case it would require too big of a refactor, the stable alternative here would be to catch the click. If the picker is unavailable, it will add the auto-wait time to the test but never return a false negative. The default action timeout is 10s, so we can also shorten it for faster execution. Here's an example implementation:

```suggestion
		await editorProductPicker
			.locator( 'label' )
			.filter( {
				hasText: 'Album',
			} )
			.click( { timeout: 3000 } )
			.catch( () => null );
	}
```

I'd also recommend changing the utility name to `chooseProductInEditorProductPickerIfAvailable` for clarity.

Reply:
> I'd also recommend changing the utility name to chooseProductInEditorProductPickerIfAvailable for clarity.

Thank you for the suggestion, I agree that changing the utility name to `chooseProductInEditorProductPickerIfAvailable` would improve clarity. I will make the changes 🤝

Reply:
Made the suggested changes in [f2284c8](https://github.com/woocommerce/woocommerce/pull/50164/commits/f2284c8e7924b82090efc4211c3aec9b47ea6aa3)